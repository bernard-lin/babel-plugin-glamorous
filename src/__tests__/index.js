import pluginTester from 'babel-plugin-tester'
import plugin from '../'

const simpleCode = `
  import g from 'glamorous'
  const MyDiv = g.div()
`

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {filename: '/path/to/SomeComponent.js'},
  tests: {
    'does not affect non-named components': {
      code: `glamorous.button();`,
      snapshot: false,
    },
    'does not affect non-glamorous imports': {
      code: `
        import styled from 'styled-components';
        style.div\`\`;
      `,
      snapshot: false,
    },
    'does not affect unused glamorous requires': {
      code: `const g = require('glamorous');`,
      snapshot: false,
    },
    'adds a display name for a component with a variable name': `
      import g from 'glamorous'
      const Button = g.button()
    `,
    'works with requires': `
      const l = require('glamorous')
      const MyDiv = l.div()
    `,
    'works with exported components': `
      import g from 'glamorous'
      export const MySection = g.section()
    `,
    'does not fail when destructuring built-in components': `
      import g from 'glamorous'
      const { Div } = g
    `,
    'handles style args': `
      import g from 'glamorous'
      const MySection = g.section({color: 'blue'}, () => ({}))
    `,
    'handles custom components': `
      import g from 'glamorous'
      const MyComp = g(MyThing, {rootEl: 'div'})()
    `,
    'does not insert "unknown" as part of the displayName': {
      babelOptions: {filename: 'unknown'},
      code: simpleCode,
    },
    'uses the parent directory if the filename is "index"': {
      babelOptions: {filename: '/foo/bar/component/index.js'},
      code: simpleCode,
    },
    'does not use the filename if it is the same as the component name': {
      babelOptions: {filename: '/foo/bar/MyDiv/index.js'},
      code: `
        import g from 'glamorous'
        const MyDiv = g.div()
      `,
    },
    'if component name cannot be inferred, it uses the filename only': {
      babelOptions: {filename: '/foo/bar/my-div.js'},
      code: `
        import g from 'glamorous'
        g.div()
      `,
    },
    'works in an object assignment': `
      import glamorous from 'glamorous'
      const comps = {
        Div: glamorous.div()
      }
    `,
    'works in member expressions': `
      import glamorous from 'glamorous'
      const comps = {}
      comps.Div = glamorous.div()
    `,
  },
})
