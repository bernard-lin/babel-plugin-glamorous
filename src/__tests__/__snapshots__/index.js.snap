// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds a display name for a component with a variable name 1`] = `
"
import g from 'glamorous'
const Button = g.button()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const Button = g.button.withConfig({
  displayName: 'SomeComponent__Button'
})();
"
`;

exports[`does not fail when destructuring built-in components 1`] = `
"
import g from 'glamorous'
const { Div } = g

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const { Div } = g;
"
`;

exports[`does not insert "unknown" as part of the displayName 1`] = `
"
import g from 'glamorous'
const MyDiv = g.div()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const MyDiv = g.div.withConfig({
  displayName: 'MyDiv'
})();
"
`;

exports[`does not use the filename if it is the same as the component name 1`] = `
"
import g from 'glamorous'
const MyDiv = g.div()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const MyDiv = g.div.withConfig({
  displayName: 'MyDiv'
})();
"
`;

exports[`handles custom components 1`] = `
"
import g from 'glamorous'
const MyComp = g(MyThing, {rootEl: 'div'})()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const MyComp = g(MyThing, { rootEl: 'div' }).withConfig({
  displayName: 'SomeComponent__MyComp'
})();
"
`;

exports[`handles style args 1`] = `
"
import g from 'glamorous'
const MySection = g.section({color: 'blue'}, () => ({}))

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const MySection = g.section.withConfig({
  displayName: 'SomeComponent__MySection'
})({ color: 'blue' }, () => ({}));
"
`;

exports[`if component name cannot be inferred, it uses the filename only 1`] = `
"
import g from 'glamorous'
g.div()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
g.div.withConfig({
  displayName: 'my-div'
})();
"
`;

exports[`uses the parent directory if the filename is "index" 1`] = `
"
import g from 'glamorous'
const MyDiv = g.div()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
const MyDiv = g.div.withConfig({
  displayName: 'component__MyDiv'
})();
"
`;

exports[`works in an object assignment 1`] = `
"
import glamorous from 'glamorous'
const comps = {
  Div: glamorous.div()
}

      ↓ ↓ ↓ ↓ ↓ ↓

import glamorous from 'glamorous';
const comps = {
  Div: glamorous.div.withConfig({
    displayName: 'SomeComponent__Div'
  })()
};
"
`;

exports[`works in member expressions 1`] = `
"
import glamorous from 'glamorous'
const comps = {}
comps.Div = glamorous.div()

      ↓ ↓ ↓ ↓ ↓ ↓

import glamorous from 'glamorous';
const comps = {};
comps.Div = glamorous.div.withConfig({
  displayName: 'SomeComponent'
})();
"
`;

exports[`works with exported components 1`] = `
"
import g from 'glamorous'
export const MySection = g.section()

      ↓ ↓ ↓ ↓ ↓ ↓

import g from 'glamorous';
export const MySection = g.section.withConfig({
  displayName: 'SomeComponent__MySection'
})();
"
`;

exports[`works with requires 1`] = `
"
const l = require('glamorous')
const MyDiv = l.div()

      ↓ ↓ ↓ ↓ ↓ ↓

const l = require('glamorous');
const MyDiv = l.div.withConfig({
  displayName: 'SomeComponent__MyDiv'
})();
"
`;
